index.html

```diff
@@ -232,39 +235,393 @@ <h3>Create</h3>
+            <li>
+Set the <em>verification method</em> `id` value by concatenating
+<em>identifier</em>, a hash character (`#`), and the <em>multicodec value</em>.
+            </li>
+            <li>
+Set the <em>key encoding type</em> value to the <em>options</em> `type` value
+if it is set, or `Multikey` if it is not set.
+            </li>
+            <li>
+If <em>options</em> `allowExperimentalPublicKeyTypes` is not set to `true` and
+<em>key encoding type</em> is not `Multikey`, `JsonWebKey2020`, or
+`Ed25519VerificationKey2020`, an INVALID_PUBLIC_KEY_TYPE error MUST be raised.
+            </li>
+            <li>
+Set the <em>verification method</em> `type` to the <em>key encoding type</em>.
+            </li>
+            <li>
+Set the <em>verification method</em> `controller` to <em>identifier</em>.
+            </li>
+            <li>
+If <em>key encoding type</em> is `Multikey` or `Ed25519VerificationKey2020`,
+set the <em>verification method</em> `publicKeyMultibase` value to
+<em>multibase value</em>.
+            <li>
+If <em>key encoding type</em> is `JsonWebKey2020`, set the
+<em>verification method</em> `publicKeyJwk` value to the result of
+passing <em>multicodec value</em> and <em>raw public key bytes</em> to
+<a href="#encode-jwk-algorithm"></a>.
+            </li>
+            <li>
+Return <em>verification method</em>.
+            </li>
+          </ol>
+
+        </section>
+
+        <section class="normative">
+          <h4>Decode Public Key Algorithm</h4>
+
+          <p>
+The following algorithm can be used for expanding a multibase-encoded
+multicodec value to decoded public key components. A <em>multibase value</em>
+and <em>options</em> are the required inputs. A <em>decoded public key</em> is
+the output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Set <em>decoded public key</em> to an empty object.
+            </li>
+            <li>
+Decode <em>multibase value</em> using the base58-btc multibase alphabet and set
+<em>multicodec value</em> to the multicodec header for the decoded value.
+Implementers are cautioned to ensure that the <em>multicodec value</em>
+is set to the result after performing varint decoding.
+            </li>
+            <li>
+Set the <em>raw public key bytes</em> to the bytes remaining after the
+multicodec header.
+            </li>
+            <li>
+Return <em>multicodec value</em> and <em>raw public key bytes</em> as the
+<em>decoded public key</em>.
+            </li>
+          </ol>
+        </section>
+
+        <section class="normative">
+          <h4>Encode JWK Algorithm</h4>
+
+          <p>
+The following algorithm can be used to encode a multicodec value to a
+JSON Web Key. A <em>multicodec value</em>, <em>raw public key bytes</em>,
+and <em>options</em> are the required inputs. An <em>encoded JWK</em> is
+the output.
+          </p>
+
+          <p class="issue">
+Add the various was of transforming multicodec values and raw public key bytes
```

Comment:
Its pretty straight forward, do you not support this today?

basically decode the public key as you do for multibase, then expand it if its expandable (secp256k1, secp256r1, 384r1, 521r1...)

encode the expanded values with `kty` `crv` `x` and `y`.... best to test this backwards and forwards from a library that generated a JWK first... instead of from a library that generated multibase first.

Reply:
> Its pretty straight forward, do you not support this today?

No, our libraries do not support JWK key expression, AFAIK.

> basically decode the public key as you do for multibase, then remove EC point compression.
> 
> encode the expanded values with `kty` `crv` `x` and `y`.... best to test this backwards and forwards from a library that generated a JWK first... instead of from a library that generated multibase first.

Sure, but the details matter here (like you have to switch off of the multicodec value, and X25519 differs slightly from other digital signing keys)... so this section has to be very explicit about the translation from multikey to JWK. I didn't want to write something that folks doing JWK stuff would disagree with. Ideally, we'd have someone that has implemented the JWK stuff write this part of the algorithm, which is why I'd rather have it done as another PR.

Reply:
We can leave it as an issue, especially if your implementations don't support this, I expect we or other JWK users will need to define this.